src/iOS.Core/Controllers/LockPasswordViewController.cs

```diff
@@ -185,9 +185,13 @@ public override async void ViewDidAppear(bool animated)
             base.ViewDidAppear(animated);
 
             // Users with key connector and without biometric or pin has no MP to unlock with
-            if (_usesKeyConnector && (!(_pinLock || _biometricLock)) || ( _biometricLock && !_biometricIntegrityValid))
+            if (_usesKeyConnector)
             {
-                PromptSSO();
```

Comment:
I'm not sure which is the standard here, but maybe these bools combinations could be properties in order to improve readability:
`bool IsLocked => _pinLock || _biometricLock;`
`bool IsBiometricLockInvalid => _biometricLock && !_biometricIntegrityValid`

```c#
if(!IsLocked || IsBiometricLockInvalid)
{
     PromptSSO();
}

```
Aside from this, I think it's ok

Reply:
I'm hesitant to add extra bool properties because they might be a little too specific. They would have to be
`bool NotUsingPinOrBio` and `bool IsBiometricLockInvalid` which seem to be describing the conditional in almost the same amount of characters. If we go this route I would also want to refactor the entire class to use them and that is a little scary since we're picking this to `rc`

Reply:
Yes, makes sense. Just in case I wasn't aming to reduce the amount of chars, only readability since reading a bunch of "||" and "&" can be hard to follow sometimes (not this particular case), but here I believe you have a point and can be left as is 😄